### ISSUE TITLE
Segmentation fault with RooTracker file

### ISSUE THREAD

**Issue by soleti** [ClarkMcGrew/edep-sim#20]
Hello,

I am trying to generate a sample of cosmic rays with `edep-sim` starting from a CORSIKA file. I wrote a converter that creates a RooTracker file starting from the CORSIKA output and I feed it to `edep-sim` using this macro:

```
/generator/kinematics/rooTracker/input DAT000001.root
/generator/kinematics/set rooTracker
/generator/position/set free
/generator/add
```
However, I get a `Segmentation fault` so I am not sure how to debug it further. You can find the DAT000001.root file [here](https://drive.google.com/file/d/1Vj_RKzMi8kjVGSabPgAWvk-vxVEFDTnR/view?usp=sharing).

Thank you,
Roberto

**Comment by ClarkMcGrew** [comment]
I've downloaded the file if you want to delete it.   I notice that it's missing a lot of the rooTracker branches.  The edep-sim rootracker code assumes that all the branches are present, so I'll add traps for that (there may still be other problems).

**Comment by soleti** [comment]
Hi Clark, thank you, I didn't know all the branches were compulsory. Adding them indeed fixes the issue (for now, I still haven't verified if the output makes sense).

**Comment by ClarkMcGrew** [comment]
Sorry for the radio silence.  I've just pushed an improved version of the EDepSimRooTrackerKinemGenerator class to master.  This adds error traps while setting the branch addresses so it should catch errors sooner.  It will fail if there is an error with a "critical" branch, but will continue otherwise.  I'm closing the since I think this fixes the fundamental problem (failing without an error message).  The fix is in 15181961.
